lib/puppet/resource.rb

```diff
@@ -88,6 +88,9 @@ def to_data_hash(json_serializer = nil)
         value = Puppet::Resource.value_to_pson_data(value)
         if is_json_type?(value)
           params[param] = value
+        elsif json_serializer.nil?
+          Puppet.warning("Resource '#{to_s}' contains a #{value.class.name} value. It will be converted to the String '#{value}'")
```

Comment:
I am not sure that is true for all serialization formats - think PSON may allow symbols